### DISCUSSION TITLE
Upgrade GCC for ESP32-WROOM-32

### DISCUSSION THREAD

**Discussion by FrancescoCrevatin** [espressif/arduino-esp32#7498] [Q&A]
Hello everybody.

I'm having a problem running a code on my ESP32, which errors are multiple and are almost identical (only the variable names change): `'audio_tools::mime_pcm' declared as an 'inline' variable
 INLINE_VAR const char* mime_pcm = "audio/pcm";`.
From what I've seen on the internet I need to upgrade the GCC (at the moment I have the 5.2 version), but I really don't know what to do. I've looked at previous topics and Espressif docs, but I don't have a clue what I have to do.

If you can give me a hand I will thank you forever.

**Comment by DeepDreamer89** [comment]
I have exact the same problem. I "solved" it with adding the following code before the first occurrence of the error:

```
#undef INLINE_VAR
#define INLINE_VAR
```

 In my case this was in line 13 in **AudioTypes.h** (in {Your_place}Arduino\libraries\arduino-audio-tools-main\src\AudioTools), so I added the code above in **line 9**.
So this error gone away:

> C:\Users\{username}\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioTools/AudioTypes.h:13:38: error: 'audio_tools::RxTxModeNames' declared as an 'inline' variable
>  INLINE_VAR const char* RxTxModeNames[]={"UNDEFINED_MODE","TX_MODE","RX_MODE","RXTX_MODE" };

I don't know what problems this may cause, but for me it solved the problem, the code now compiles and run...

**Comment by Jason2866** [comment]
you can not easily use a newer gcc version, since IDF 4.4 is used to compile the needed Arduino libs.
IDF 4.4 code is not ready to use newer gcc versions.
